### ISSUE TITLE
Removing the transformations in the validation phase

### ISSUE THREAD

**Issue by denadai2** [victoresque/pytorch-template#62]
Hi,
I would like to remove some of the transformations in the data loader when there is the validation phase. Any suggestion to do it in a clean way through your template?

thanks

**Comment by SunQpark** [comment]
I think applying different transforms is easiest when you have separate datasets, but unfortunately `BaseDataLoader` class in this template is not suitable for this use case. Instead, you can use `torch.utils.data.random_split` like below example.

```python
from torch.utils.data import DataLoader, random_split

def get_datasets(data_dir, batch_size, validation_split=0.0, num_workers=1):
    train_tr = transforms.Compose([
        transforms.ToTensor(),
        transforms.Normalize((0.1307,), (0.3081,))
    ])
    valid_tr = transforms.ToTensor()

    dataset = datasets.MNIST(data_dir, transform=train_tr)
    train_dset, valid_dset = random_split(dataset, len(dataset) * validation_split) # split dataset into train / valid
    valid_dset.transform = valid_tr # replace transform for valid set 

    train_loader = DataLoader(train_dset, batch_size, shuffle=True, num_workers=num_workers)
    valid_loader = DataLoader(valid_dset, batch_size, shuffle=True, num_workers=num_workers)
    return train_loader, valid_loader
```

**Comment by SunQpark** [comment]
train / valid splitting of `BaseDataLoader` in this project, or `torch.utils.data.random_split` only split the indices of dataset. I think the best method of doing this is to have different dataset object for train and validation set. 
1. modify your dataset class to have `training=True` in `__init__`, and use different data by this flag.
2. initialize `train_dataset` and `valid_dataset` respectively. 
3. Wrap them with `torch.utils.data.DataLoader`, to make them `train_loader` and `valid_loader`
4. override the `split_validation` method to just return `valid_loader`

**Comment by denadai2** [comment]
Thank you
